src/output/BaseOutputSettler.sol

```diff
@@ -169,29 +166,69 @@ abstract contract BaseOutputSettler is IPayloadCreator, BaseOracle {
     // --- IPayloadCreator --- //
 
     /**
-     * @notice Helper function to check whether a payload hash is valid.
-     * @param payloadHash keccak256 hash of the relevant payload.
+     * @notice Helper function to check whether a payload is valid.
+     * @dev Works by checked if the entirety of the payload has been recorded as valid. Every byte of the payload is
+     * checked to ensure the payload has to be filled.
+     * @param payload keccak256 hash of the relevant payload.
      * @return bool Whether or not the payload has been recorded as filled.
      */
     function _isPayloadValid(
-        bytes32 payloadHash
+        bytes calldata payload
     ) internal view virtual returns (bool) {
-        return _attestations[block.chainid][bytes32(uint256(uint160(address(this))))][bytes32(
-            uint256(uint160(address(this)))
-        )][payloadHash];
+        // Check if the payload is large enough for it to be a fill description.
+        if (payload.length < 168) revert PayloadTooSmall();
+        bytes32 outputHash = MandateOutputEncodingLib.getMandateOutputHashFromCommonPayload(
+            bytes32(uint256(uint160(msg.sender))), // Oracle
```

Comment:
is `msg.sender` ok here? shouldn't this be `address(this)`  ?

also see https://github.com/openintentsframework/oif-contracts/pull/12/files#r2172909062